From b918801901bd4c8d6f7ad60d9325ea4b3b895a13 Mon Sep 17 00:00:00 2001
From: Tristan Cacqueray <tdecacqu@redhat.com>
Date: Wed, 17 May 2017 12:36:00 +0000
Subject: [PATCH] sql-reporter: add support for Ref change

This change fixes AttributeErrors when the sql reporter
is used in post/periodic pipelines where the item.change is of
type Ref without number/patchset/refspec members.

Change-Id: I11696b41a0a71405dfb7845ce222ef302a96986b
---

diff --git a/zuul/reporter/sql.py b/zuul/reporter/sql.py
index b663a59..86ac995 100644
--- a/zuul/reporter/sql.py
+++ b/zuul/reporter/sql.py
@@ -46,13 +46,25 @@
             if 'score' in self.reporter_config else 0
 
         with self.connection.engine.begin() as conn:
+            if hasattr(item.change, 'number'):
+                change = item.change.number
+            else:
+                change = ''
+            if hasattr(item.change, 'patchset'):
+                patchset = item.change.patchset
+            else:
+                patchset = ''
+            if hasattr(item.change, 'refspec'):
+                refspec = item.change.refspec
+            else:
+                refspec = item.change.newrev
             buildset_ins = self.connection.zuul_buildset_table.insert().values(
                 zuul_ref=item.current_build_set.ref,
                 pipeline=item.pipeline.name,
                 project=item.change.project.name,
-                change=item.change.number,
-                patchset=item.change.patchset,
-                ref=item.change.refspec,
+                change=change,
+                patchset=patchset,
+                ref=refspec,
                 score=score,
                 message=self._formatItemReport(
                     pipeline, item, with_jobs=False),
